mist-core/src/main/java/edu/snu/mist/core/task/groupaware/GroupAwareQueryManagerImpl.java

```diff
@@ -176,7 +191,12 @@ public QueryControlResult create(final AvroDag avroDag) {
       }
 
       final ApplicationInfo applicationInfo = applicationMap.get(appId);
-      final DAG<ConfigVertex, MISTEdge> configDag = configDagGenerator.generate(avroDag);
+      final DAG<ConfigVertex, MISTEdge> configDag;
+      if (checkpointedState == null) {
+        configDag = configDagGenerator.generate(avroDag);
+      } else {
+        configDag = configDagGenerator.generateWithCheckpointedStates(avroDag, checkpointedState);
```

Comment:
Could you please explain how `PhysicalOperator` restores the state?

Reply:
If a `ConfigVertex` has its internal states, it is deserialized and recovered in `ExecutionDagGenerator`.